apiserver/Monitoring.md

```diff
@@ -0,0 +1,71 @@
+# Monitoring of the API server and created Ray clusters
+
+In order to ensure a proper functioning of the API server and created Ray clusters, it is typically necessary to
+monitor the API server itself and created Ray clusters. This document describes how to instrument both API server
+and created clusters with Prometheus and Grafana
+
+## Monitoring of the API server
+
+Current implementation of the API server provides a flag `collectMetricsFlag` that defines whether to collect and
+expose Prometheus metrics, which is by default set to `true`. To disable metrics collection, this flag needs to
+be set to `false` and the API server image needs to be rebuild.
+
+If this flag is enabled, an `http` port at `/metrics` endpoint provides API server metrics in Prometheus format.
+
+### Testing monitoring of the API server
+
+On the kind server Install Kubernetes Prometheus Stack via Helm chart using the following
+[command](../install/prometheus/install.sh). The script additionally creates `ray-head-monitor` and
+`ray-workers-monitor` in the `prometheus-system` namespace, that we do not need. We can delete them using:
+
+```shell
+kubectl delete servicemonitor ray-head-monitor -n prometheus-system
+kubectl delete podmonitor ray-workers-monitor -n prometheus-system
+```
+
+Install API server from the project using the following command:
+
+```shell
+helm install apiserver /Users/boris/Projects/kuberay/helm-chart/kuberay-apiserver
+```
+
+Now we can install a [service monitor](test/api_server_service_monitor.yaml) to scrape Api Server metrics into
```

Comment:
See my comment on this file.

Reply:
???

Reply:
I was refering to the `test/api_server_service_monitor.yaml` being a duplicate of the `apiserver/deploy/base/monitor.yaml` file

Reply:
Fixed